components/org.wso2.carbon.identity.sso.saml/src/main/java/org/wso2/carbon/identity/sso/saml/util/SAMLSSOUtil.java

```diff
@@ -2580,6 +2582,16 @@ private static String getSPQualifier(QueryParamDTO[] queryParamDTOs) {
         return null;
     }
 
+    /**
+     * Resolves the public service url given the default context and the url picked from the configuration based on
+     * the 'tenant_context.enable_tenant_qualified_urls' mode set in deployment.toml.
+     *
+     * @param defaultUrlContext default url context path
+     * @param urlFromConfig     url picked from the file configuration
+     * @return absolute public url of the service if 'enable_tenant_qualified_urls' is 'true', else returns the url
+     * from the file config
+     * @throws URLBuilderException when fail to build the absolute public url
+     */
     private static String resolveUrl(String defaultUrlContext, String urlFromConfig) throws URLBuilderException {
 
         if (!IdentityTenantUtil.isTenantQualifiedUrlsEnabled() && StringUtils.isNotBlank(urlFromConfig)) {
```

Comment:
when isTenantQualifiedUrlsEnabled() = false and the urlFromConfig is empty, shouldn't this return the url built from defaultUrlContext?

Reply:
it means this condition becomes false right.
So it builds the URL from default context as expected at line 2605